### ISSUE TITLE
Error: TrustFailure (Authentication failed, see inner exception.) on Debian

### ISSUE THREAD

**Issue by kolAflash** [KSP-CKAN/CKAN#3915]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Operating System

Debian-12

### CKAN Version

v1.33.2

### Game Version

0.1.4.0

### Describe the bug

When trying to download a mod via CKAN reports:
```
One or more downloads were unsuccessful:
Error downloading XXX Y.Y.Y: The remote server returned an error: (404) Not Found.
```
Tested with GUI, consoleui and console `install` command.

&nbsp;

Running the GUI with --verbose I get this via STDOUT/STDERR:
```
60413 [Thread Pool Worker] INFO CKAN.NetAsyncDownloader (null) - Error downloading https://spacedock.info/mod/3270/Maneuver Node Controller/download/1.0.3: Error: TrustFailure (Authentication failed, see inner exception.)
60413 [Thread Pool Worker] INFO CKAN.NetAsyncDownloader (null) - Trying fallback URL: https://archive.org/download/ManeuverNodeController-1.0.3/A08D85DE-ManeuverNodeController-1.0.3.zip
60425 [Thread Pool Worker] INFO CKAN.NetAsyncDownloader (null) - Setting MIME type application/zip;q=1.0,application/octet-stream;q=0.9
63371 [Thread Pool Worker] INFO CKAN.NetAsyncDownloader (null) - Error downloading https://spacedock.info/mod/3270/Maneuver Node Controller/download/1.0.3: The remote server returned an error: (404) Not Found.
63371 [Thread Pool Worker] INFO CKAN.NetAsyncModulesDownloader (null) - The remote server returned an error: (404) Not Found.
```
When copying `https://spacedock.info/mod/3270/Maneuver Node Controller/download/1.0.3` to a browser the download is successful.

&nbsp;

I've tested several mods and it's always the same problem.

My best guess is, that the spaces `" "` in the url are not properly converted to `%20`. Maybe due to some difference in the http/url implementation in Linux Mono...  
https://en.wikipedia.org/wiki/Percent-encoding#Percent-encoding_reserved_characters

&nbsp;

&nbsp;

P.S.  
Many mods use [SpaceWarp](https://github.com/SpaceWarpDev/SpaceWarp) require BepInEx. But BepInEx needs a quirk running in Proton on Linux. It's pretty straight forward. But when installing via CKAN there's nothing giving a hint to configure that quirk. So a user might just wonder, why mods do not work.  
https://docs.bepinex.dev/articles/advanced/proton_wine.html
An easy solution would be showing a message when running on Linux, advising the user to read that web page.  
Ideally CKAN would apply that quirk automatically. But I guess that would be pretty complicated.

**Comment by HebaruSan** [comment]
"Error: TrustFailure (Authentication failed, see inner exception.)" means it's probably your certificates:

https://github.com/KSP-CKAN/CKAN/wiki/SSL-certificate-errors

**Comment by kolAflash** [comment]
Indeed, it seems there was a certificate missing. Running `sudo update-ca-certificates` seems to work too in Debian-12.

**Comment by HebaruSan** [comment]
Glad you got it working! 🎉

**Comment by kolAflash** [comment]
I'm wondering why I had to do this manually. Shouldn't the OS keep the certificates up to date?

The `ca-certificates` package was updated when I upgraded from Debian-11 to Debian-12 at 2023-04-28. At that point `ca-certificates` should have been reconfigured and updated that certificates.

I guess those are root certificates. As far as I understand there should be no need to update them more often than once in every few years...!?
